deployment/security-privacy.mdx

```diff
@@ -36,3 +37,66 @@ you use [dedicated infrastructure](/deployment/dedicated-infrastructure) for pro
 We monitor the public relays for abuse and malicious activity. If we detect
 abuse, we reserve the right to block offending IP addresses or users from
 accessing the public relays.
+
+
+## Protecting leakage of IP Addresses 
+
+Any direct connection between two devices stands a high chance of being the fastest
+connection, but always requires IP addresses to be disclosed. As with the majority
+of the deployed internet stack today, when two endpoints establish a direct
+connection, they expose and exchange their IP addresses to each other.
+
+IP address privacy considerations are primarily relevant for consumer or peer-to-peer applications
+where strangers or untrusted parties connect directly over the internet. In these
+scenarios, developers cannot guarantee anonymity or trust between endpoints, and
+exposing IP addresses can lead to privacy risks such as location tracking or
+targeted attacks.
+
+To mitigate these risks, we recommend the following strategies to protect IP
+address privacy when using iroh, depending on your specific use case and threat model:
+
+### Basic Protection: Use dedicated infrastructure
+
+If you own the infrastructure, run your own DNS server, and
+manage the devices connecting to your network, you have control over the network
+topology and can implement appropriate security measures. We do not recommend using
+public relays for production systems, as this is shared public
+infrastructure that has no guarantees around privacy or uptime.
+
+We recommend reviewing our [dedicated
+infrastructure](/deployment/dedicated-infrastructure) guidance to set up relay
+and DNS infrastructure that fits your needs.
+
+### More protection: Use Tor or Similar Onion Routing
+
+Services like Tor provide onion routing, which encrypts packet metadata for each
+relay in the route. This offers strong IP privacy guarantees through multi-hop
+routing with layered encryption.
+
+### Upcoming
+
+#### Relay-Only Mode 
+
+Upcoming releases of iroh will support disabling hole-punching to send all traffic
+exclusively through relays. This provides IP privacy with some important caveats:
+
+- **Single-hop routing**: Traffic passes through one relay, not multiple hops
+- **Trust required**: The relay operator can technically see which endpoints are
+  communicating and their IP addresses
+- **Data remains encrypted**: The relay cannot read the actual content due to
+  end-to-end encryption
+
+This mode is suitable when you trust your relay infrastructure but still want to
+avoid direct IP exposure between endpoints.
+
+#### Multi-Hop Relay Routing
+
+We've explored the possibility of adding multi-hop relay routing to iroh. While
+technically feasible, this feature is not currently on the near-term roadmap.
```

Comment:
I was scared seeing this heading, but immediately soothed after reading the first sentence :joy: